Desktop/QueryRetrieve SCU/Program.cs

```diff
@@ -0,0 +1,239 @@
+
+            // now get all the images of a serie with cGet in the same association
+
+            var cGetRequest = CreateCGetBySeriesUID(studyUID, serieUids.First());
+            client.OnCStoreRequest += (DicomCStoreRequest req) =>
+            {
+                SaveImage(req.Dataset);
+                return new DicomCStoreResponse(req, DicomStatus.Success);
+            };
+            // the client has to accept storage of the images. We know that the requested images are of SOP class Secondary capture, 
+            // so we add the Secondary capture to the additional presentation context
+            // a more general approach would be to mace a cfind-request on image level and to read a list of distinct SOP classes of all
+            // the images. these SOP classes shall be added here.
+            client.AdditionalPresentationContexts.Add(DicomPresentationContext.GetScpRolePresentationContext(
+                DicomUID.SecondaryCaptureImageStorage, DicomTransferSyntax.ImplicitVRBigEndian,
+                DicomTransferSyntax.ExplicitVRBigEndian, DicomTransferSyntax.ExplicitVRLittleEndian));
+            client.AddRequest(cGetRequest);
+            client.Send(QRServerHost, QRServerPort, false, AET, QRServerAET);
+
+            // if the images shall be sent to an existing storescp and this storescp is configured on the QR SCP then a CMove could be performed:
+
+            // here we want to see how a error case looks like - because the test QR Server does not know the node FODICOMSCP
+            var cMoveRequest = CreateCMoveBySeriesUID("FODICOMSCP", studyUID, serieUids.First());
+            bool? moveSuccessfully = null;
+            cMoveRequest.OnResponseReceived += (DicomCMoveRequest requ, DicomCMoveResponse response) =>
+            {
+                if (response.Status.State == DicomState.Pending)
+                {
+                    Console.WriteLine("Sending is in progress. please wait");
+                }
+                else if (response.Status.State == DicomState.Success)
+                {
+                    Console.WriteLine("Sending successfully finished");
+                    moveSuccessfully = true;
+                }
+                else if (response.Status.State == DicomState.Failure)
+                {
+                    Console.WriteLine("Error sending datasets: " + response.Status.Description);
+                    moveSuccessfully = false;
+                }
+                Console.WriteLine(response.Status);
+            };
+            client.AddRequest(cMoveRequest);
+            client.Send(QRServerHost, QRServerPort, false, AET, QRServerAET);
+
+            if (moveSuccessfully.GetValueOrDefault(false))
+            {
+                // images sent successfully from QR Server to the store scp
+            }
+            Console.ReadLine();
+        }
+
+
+        public static DicomCFindRequest CreateStudyRequestByPatientName(string patientName)
```

Comment:
Couldn't you consider the library method `DicomCFindRequest.CreateStudyQuery` instead?